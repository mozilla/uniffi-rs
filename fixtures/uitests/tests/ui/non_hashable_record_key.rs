// Unfortunately, path is relative to a temporary build directory :-/
uniffi_macros::generate_and_include_scaffolding!("../../../../fixtures/uitests/src/records.udl");

fn main() { /* empty main required by `trybuild` */}
