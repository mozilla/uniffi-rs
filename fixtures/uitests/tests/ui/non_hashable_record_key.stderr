error[E0277]: the trait bound `f32: std::cmp::Eq` is not satisfied
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | uniffi::deps::static_assertions::assert_impl_all!(f32: ::std::cmp::Eq, ::std::hash::Hash); // record<f32, u64>
  |                                                   ^^^ the trait `std::cmp::Eq` is not implemented for `f32`
  |
  = help: the following other types implement trait `std::cmp::Eq`:
            isize
            i8
            i16
            i32
            i64
            i128
            usize
            u8
          and $N others
note: required by a bound in `assert_impl_all`
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | uniffi::deps::static_assertions::assert_impl_all!(f32: ::std::cmp::Eq, ::std::hash::Hash); // record<f32, u64>
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the macro `uniffi::deps::static_assertions::assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f32: Hash` is not satisfied
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | uniffi::deps::static_assertions::assert_impl_all!(f32: ::std::cmp::Eq, ::std::hash::Hash); // record<f32, u64>
  |                                                   ^^^ the trait `Hash` is not implemented for `f32`
  |
  = help: the following other types implement trait `Hash`:
            isize
            i8
            i16
            i32
            i64
            i128
            usize
            u8
          and $N others
note: required by a bound in `assert_impl_all`
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | uniffi::deps::static_assertions::assert_impl_all!(f32: ::std::cmp::Eq, ::std::hash::Hash); // record<f32, u64>
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the macro `uniffi::deps::static_assertions::assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f32: Hash` is not satisfied
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `f32`, which is required by `HashMap<f32, u64>: LowerReturn<UniFfiTag>`
  |
  = help: the following other types implement trait `Hash`:
            isize
            i8
            i16
            i32
            i64
            i128
            usize
            u8
          and $N others
  = note: required for `HashMap<f32, u64>` to implement `Lower<UniFfiTag>`
  = note: required for `HashMap<f32, u64>` to implement `LowerReturn<UniFfiTag>`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f32: std::cmp::Eq` is not satisfied
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `f32`, which is required by `HashMap<f32, u64>: LowerReturn<UniFfiTag>`
  |
  = help: the following other types implement trait `std::cmp::Eq`:
            isize
            i8
            i16
            i32
            i64
            i128
            usize
            u8
          and $N others
  = note: required for `HashMap<f32, u64>` to implement `Lower<UniFfiTag>`
  = note: required for `HashMap<f32, u64>` to implement `LowerReturn<UniFfiTag>`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find function `get_dict` in this scope
 --> $OUT_DIR[uniffi_uitests]/records.uniffi.rs
  |
  | pub fn r#get_dict(
  |        ^^^^^^^^^^ not found in this scope
